packages/checkout/widgets-lib/src/widgets/wallet/components/WalletAddress/WalletAddress.tsx

```diff
@@ -19,15 +62,47 @@ export function WalletAddress({ provider }: { provider?: Web3Provider }) {
     })();
   }, [provider]);
 
+  const handleIconClick = () => {
+    if (walletAddress && ctaIcon() === 'CopyText') {
+      track({
+        userJourney: UserJourney.WALLET,
+        screen: 'Settings',
+        control: 'CopyWalletAddress',
+        controlType: 'Button',
+      });
+      navigator.clipboard.writeText(walletAddress);
+
+      setIsCopied(true);
+      setTimeout(() => {
+        setIsCopied(false);
+      }, 1000);
+    } else if (ctaIcon() === 'ShowPassword') {
+      setShowL1Warning(true);
+    }
+  };
+
   return (
-    <Box sx={walletAddressContainerStyle}>
-      <Box sx={walletAddressLayoutStyle}>
-        <Box sx={walletAddressTextLayoutStyle}>
-          <Heading size="xSmall">Wallet address</Heading>
-          <Body testId="wallet-address" size="xSmall" sx={truncatedTextStyle}>{walletAddress}</Body>
-        </Box>
-        <CopyButton textToCopy={walletAddress} />
-      </Box>
-    </Box>
+    <MenuItem testId="wallet-address" emphasized size="medium">
+      <MenuItem.FramedLogo
+        logo={getWalletLogoByName(walletProviderName())}
```

Comment:
Can we move the call to `getWalletLogoByName` out of the template to wherever `walletProviderName` gets moved to?

Reply:
replacing `walletProviderName` with `getWalletProviderNameByProvider` as we already have it